exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/traces/SpanMarshaler.java

```diff
@@ -59,6 +58,10 @@ static SpanMarshaler create(SpanData spanData) {
             ? EMPTY_BYTES
             : encodeTraceState(traceState).getBytes(StandardCharsets.UTF_8);
 
+    int flags =
+        ProtoSpanFlagsUtil.getFullSpanFlags(
+            spanData.getSpanContext().getTraceFlags().asByte(), spanData.getParentSpanContext());
```

Comment:
Instead of losing the nice `TraceFlags` abstraction and replacing it with an int, can we instead keep the `TraceFlags` and encapsulate this in it? Might that also help to reduce the duplicated call to this in both marshalerers?

Reply:
Yeah, that was something I was thinking of. The reason I didn't want to touch TraceFlags is the following:

The current `TraceFlags` class represents only the W3C flags (the 8 least significant bits). And it also exposes this detail through it's API (methods like `getByte()`). This is the part of the flags that is also being propagated with the context.

What we need for this feature, however, is the full trace flags representation (i.e. 4 bytes) of the OTLP definition. These bits are not being propagated with the context. 

So changing the  `TraceFlags` class to represent the flags as an int instead of a byte might potentially have way more far reaching impact than just the exporting these additional bits. 

But I see there might be a way to encapsulate it in TraceFlags while keeping the previous logic (of 1 byte) untouched, WDYT?

Reply:
Encapsulated this into TraceFlags now. 
@breedx-splk PTAL